source/grid/grid_tools.cc

```diff
@@ -2888,9 +2917,21 @@ namespace GridTools
                     boost::geometry::index::satisfies(marked),
                   std::back_inserter(res));
 
-                // We should have one and only one result
-                AssertDimension(res.size(), 1);
-                closest_vertex_index = res[0].second;
+                Assert(res.size() < 2,
```

Comment:
So `res` can now have size 0 or 1? Can you elaborate in a comment what size 0 means?

Reply:
ok :)